custom_components/sector/binary_sensor.py

```diff
@@ -124,57 +121,33 @@ def __init__(
         self,
         coordinator: SectorDataUpdateCoordinator,
         serial_no: str,
-        sensor_type: str,
         device_info: dict,
-        device_class: BinarySensorDeviceClass,
+        description: GenericSensorEntityDescription,
         model: str,
     ) -> None:
         """Initialize the binary sensor."""
         super().__init__(coordinator, serial_no, device_info, model)
-        self._sensor_type = sensor_type
-        self._attr_unique_id = f"{serial_no}_{sensor_type}"
-        self._attr_name = f"{sensor_type.replace('_', ' ').capitalize()}"
-        self._attr_device_class = device_class
+        self.entity_description = description
+        self._attr_unique_id = f"{serial_no}_{description.sensor_type}"
+        self._sensor_type = description.sensor_type
```

Comment:
```suggestion
```
Seems no added value to create and use this

Reply:
Removed.